### PR TITLE
fix(entity-renderer): added null checks

### PR DESCRIPTION
**Pull request by 23nobody**
Made Entities non navigable and tooltip text to be `unknown` for the null cases.

Screenshot 2022-06-08 at 4 16 04 PM

## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { EntityNavigationService } from '../../services/navigation/entity/entity
       *ngIf="this.name"
       class="ht-entity-renderer"
       [ngClass]="{ 'default-text-style': !this.inheritTextStyle }"
-      [htTooltip]="this.name"
+      [htTooltip]="this.getTooltipText | htMemoize: this.name"
```
Comment (adisreyaj) [inline projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts:17]:
```suggestion
      [htTooltip]="this.getTooltipText | htDisplayString : 'unknown'"
```
`htDisplayString` should be taking care of `null` and `'null'` values.

Comment (23nobody) [reply projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts]: Oh, forgot about this. Thanks!
Updated.

File: projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import { EntityNavigationService } from '../../services/navigation/entity/entity
       *ngIf="this.name"
       class="ht-entity-renderer"
       [ngClass]="{ 'default-text-style': !this.inheritTextStyle }"
-      [htTooltip]="this.name"
+      [htTooltip]="this.name | htDisplayString: 'unknown'"
```
Comment (adisreyaj) [inline projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts:17]: Maybe capitalize `Unknown`?

Comment (23nobody) [reply projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts]: Updated.

File: projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts [hunk 3]
```diff
@@ -96,7 +96,7 @@ export class EntityRendererComponent implements OnChanges {
 
   private setNavigationParams(): void {
     this.navigationParams =
-      this.navigable && this.entity !== undefined
+      this.navigable && this.entity !== undefined && this.entity[entityIdKey] !== 'null'
```
Comment (anandtiwary) [inline projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts:99]: Why are we passing invalid entities?

**Review by anandtiwary — CHANGES_REQUESTED** [review]
Need to discuss

File: projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ import { EntityNavigationService } from '../../services/navigation/entity/entity
       *ngIf="this.name"
       class="ht-entity-renderer"
       [ngClass]="{ 'default-text-style': !this.inheritTextStyle }"
-      [htTooltip]="this.name"
+      [htTooltip]="this.name | htDisplayString: 'Unknown'"
```
[the remark this answers is not in the document]
Comment (23nobody) [reply projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts:17]: we can have attacks identified on requests for which we haven’t learnt the API endpoints. For that events we get `apiId = null` and 'apiName = null'

File: projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts (hunk 3 above)
Comment (23nobody) [reply projects/observability/src/shared/components/entity-renderer/entity-renderer.component.ts:99]: we can have attacks identified on requests for which we haven’t learnt the API endpoints. For that events we get apiId = null and `apiName = null`

**Comment by 23nobody** [comment]
Will need to add checks at the entry level in the UI code. And add handling accordingly in the `rich-entity-with-parent` component.

**Pull request closed without merging**